### PR TITLE
Release versioning

### PR DESCRIPTION
**Pull request by pownkel**
#### Description of changes

Save release id as an env variable in function app so we can query a specific build's test results. New script (update-release-version.sh) should be added as a new step in the build pipeline after restarting batch pools so that we don't accidentally query test runs on a system that is not fully updated.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Added relevant unit test for your changes. (`yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] Ran precheckin (`yarn precheckin`)

### CODE REVIEW

File: packages/resource-deployment/scripts/update-release-version.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+getFunctionAppNames() {
+    functionAppNamesString=$(az functionapp list --resource-group "$resourceGroupName" --query '[].name' -o tsv | tr "\n" ",")
+    echo "Found function apps: $functionAppNamesString"
+    IFS=$',' read -ra functionAppNames <<<"$functionAppNamesString"
+}
+
+setReleaseVersion() {
+    functionAppName=$1
+    echo "Updating release version for function app $functionAppName"
+    az functionapp config appsettings set --name $functionAppName --resource-group $resourceGroupName --settings "RELEASE_VERSION=$releaseVersion" 1>/dev/null
```
Comment (devaradhanm) [inline packages/resource-deployment/scripts/update-release-version.sh]: i would suggest to do this as part of the arm template. this way, it is consistent with how we set other env variables.

Comment (devaradhanm) [reply packages/resource-deployment/scripts/update-release-version.sh]: (like in function-app-template.json)

File: packages/web-api/src/controllers/health-check-controller.ts [hunk 2]
```diff
@@ -48,7 +54,7 @@ export class HealthCheckController extends ApiController {
         };
     }
 
-    // Override this method not to check api version
+    // Override thqgitis method not to check api version
```
Comment (lamaks) [inline packages/web-api/src/controllers/health-check-controller.ts]: typo

File: packages/web-api/src/controllers/health-check-controller.ts [hunk 3]
```diff
@@ -26,20 +26,26 @@ export class HealthCheckController extends ApiController {
...
             testsFailed: 0,
         };
 
-        const e2eTestConfig = await this.serviceConfig.getConfigValue('e2eTestConfig');
-        const queryString = 'customEvents | limit 5';
-        const queryResponse = await appInsightsClient.executeQuery(queryString, e2eTestConfig.testRunQueryTimespan);
-        if (queryResponse.statusCode === 200) {
-            this.logger.logInfo(`App insights api queried with result ${JSON.stringify(queryResponse)}`);
+        if (currentVersion === undefined) {
+            healthReport.error = 'No release version number is configured';
```
Comment (lamaks) [inline packages/web-api/src/controllers/health-check-controller.ts]: we probably should follow same approach we have with rest api and return HTTP 400 Bad Request with message body as we have in other apis

Comment (pownkel) [reply packages/web-api/src/controllers/health-check-controller.ts]: Do we want to do that if it's a configuration issue with the function app, not a problem with the request?

File: packages/resource-deployment/scripts/update-release-version.sh (hunk 1 above)
Comment (haonliu) [reply packages/resource-deployment/scripts/update-release-version.sh]: Are we going to do it inside the arm template?

File: packages/web-api/src/controllers/health-check-controller.ts (hunk 3 above)
Comment (lamaks) [reply packages/web-api/src/controllers/health-check-controller.ts]: if we have problem with configuration we should send HTTP 500 instead

**Pull request merged**